lib/oli_web/components/delivery/activity_helpers.ex

```diff
@@ -0,0 +1,624 @@
+defmodule OliWeb.Delivery.ActivityHelpers do
+  @moduledoc """
+  Common helper functions for rendering activities with metrics
+  in the instructor dashboard's Insights View (Scored Activities, Practice Activities and Surveys)
+  """
+
+  use OliWeb, :html
+
+  alias Oli.Analytics.Summary
+  alias Oli.Delivery.Attempts.Core
+  alias Oli.Delivery.Sections.Section
+  alias Oli.Publishing.DeliveryResolver
+  alias OliWeb.ManualGrading.RenderedActivity
+
+  def get_activities(assessment_resource_id, section_id, student_ids, filter_by_survey \\ false),
+    do:
+      Core.get_evaluated_activities_for(
```

Comment:
This is a really nice isolation of redundant code that had been spread across Practice, Summative and Survey views.  But we really should be taking this opportunity to move away from querying across ActivityAttempt and ResourceAttempt tables.   Instead, this information can be much more efficiently retrieved by querying across Summary tables and then joining to the corresponding SectionResource record (through section id and resource id) to get the title of the resource.